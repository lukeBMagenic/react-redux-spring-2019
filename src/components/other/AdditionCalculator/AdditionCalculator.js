import React, { Component, Fragment } from 'react';
import './AdditionCalculator.scss';

class AdditionCalculator extends Component {
  constructor(props) {
    super(props);
    this.state = { value1: 0, value2: 0};
  }
  
  handleValue1Change = e => {
    this.setState({ value1: e.target.value });
  }

  handleValue2Change = e => {
    this.setState({ value2: e.target.value });
  }

  render() {
    const value1 = this.state.value1;
    const value2 = this.state.value2;
    return (
      <Fragment>
        <fieldset>
          <legend>Add up some numbers</legend>
          <input value={value1} onchange={this.handleValue1Change} />
          <input value={value2} onchange={this.handleValue2Change} />
        </fieldset>
        <h1>Sum of these numbers {parseInt(value1) + parseInt(value2)}</h1>
      </Fragment>
    );
  }
}

export default AdditionCalculator;
