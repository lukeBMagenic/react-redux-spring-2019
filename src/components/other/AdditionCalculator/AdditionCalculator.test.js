import React from 'react';
import { shallow } from "enzyme";

import AdditionCalculator from './AdditionCalculator';

describe("AdditionCalculator", () => {
  it("should render correctly", () => {
    const component = shallow(<AdditionCalculator />);

    expect(component).toMatchSnapshot();
  });
});
