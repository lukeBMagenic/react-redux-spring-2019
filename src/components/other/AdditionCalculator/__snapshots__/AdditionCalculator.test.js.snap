// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionCalculator should render correctly 1`] = `
<Fragment>
  <fieldset>
    <legend>
      Add up some numbers
    </legend>
    <input
      onchange={[Function]}
      value={0}
    />
    <input
      onchange={[Function]}
      value={0}
    />
  </fieldset>
  <h1>
    Sum of these numbers 
    0
  </h1>
</Fragment>
`;
